src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb

```diff
@@ -181,6 +181,9 @@ def create(bosh_vm_meta, location, vm_props, network_configurator, env)
         end
       else
         begin
+          # Delete the empty availability set
+          _delete_empty_availability_set(resource_group_name, availability_set[:name]) if availability_set
```

Comment:
I think it's better to put all the deleting operations into seperated begin rescue block, so the one failure do not make others fail.

Reply:
These calls are independent, it is better to put them in threads.